### PR TITLE
martian: enable gosec, errcheck and bodyclose linters 

### LINKED ISSUE #360: golangci: review bodyclose linter
The bodyclose linter was disabled in 3edbdb99 due to its poor quality. 
Check if the linter improved and decide of its fate at forwarder. 

For reference: https://github.com/timakin/bodyclose/issues/52

### LINKED ISSUE #361: martian: enable gosec linter
The gosec linter was disabled in ed512616.

### LINKED ISSUE #363: martian: enable errcheck linter
The errcheck linter was disabled in c869381a.

### PR DESCRIPTION
**Pull request by mmatczuk**
Fixes #361
Fixes #363 
Fixes #360

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -108,8 +108,11 @@ issues:
     - path: internal/martian
       linters:
         - goheader
-        - errcheck
```
Comment (Choraden) [inline .golangci.yml:111]: This does not belong in that commit.

Comment (mmatczuk) [reply .golangci.yml:111]: Some rebasing must have gone wrong. Good catch

File: internal/martian/mitm/mitm.go [hunk 2]
```diff
@@ -224,6 +225,7 @@ func (c *Config) TLSForHost(ctx context.Context, hostname string) *tls.Config {
 		nextProtos = []string{"h2", "http/1.1"}
 	}
 	return &tls.Config{
+		MinVersion: tls.VersionTLS12,
```
Comment (Choraden) [inline internal/martian/mitm/mitm.go:221]: Won't it break anything for sc customers?

File: internal/martian/proxy_handler.go [hunk 3]
```diff
@@ -157,15 +157,18 @@ func (p proxyHandler) handleUpgradeResponse(rw http.ResponseWriter, req *http.Re
 	}
 }
 
-func (p proxyHandler) tunnel(name string, rw http.ResponseWriter, req *http.Request, res *http.Response, crw io.ReadWriteCloser) error {
+func (p proxyHandler) tunnel(name string, rw http.ResponseWriter, req *http.Request, res *http.Response, crw io.ReadWriteCloser) (ferr error) {
 	rc := http.NewResponseController(rw)
 
+	defer func() {
+		p.traceWroteResponse(res, ferr)
+	}()
```
Comment (Choraden) [inline internal/martian/proxy_handler.go:165]: Good!

**Review by Choraden — APPROVED** [review]
Great work!

**Pull request merged**